packages/discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -536,6 +541,7 @@ def index_user_tip(
 
 
 def process_transfer_instruction(
+    solana_client_manager: SolanaClientManager,
```

Comment:
I updated this to pass the client manager in instead of having functions further down the stack read them off the celery task to make this code more testable. We can pass in a mock version instead of needing to hook into the celery task and inject a property there. We were already doing this for all the other dependencies of this function.

Ultimately I think we probably shouldn't be reading things off the task object anywhere but the actual work function.

Reply:
agree